networking.md

```diff
@@ -0,0 +1,88 @@
+{{% /notice %}}
+
+### Establishing connections
+
+When a filecoin node connects to the network, it may connect to multiple nodes at once. When a new node connects, it must be greeted with the `HelloMessage` through the hello-handshake protocol first. Only after that message has been sent can other streams be opened. If a node receives a `HelloMessage` with a GenesisBlock it doesn't support, it must immediately close the entire connection to that peer.
+
+If the node learns through that handshake about newer blocks it should request these immediately to sync up their chain.
+
+### Syncing
+
+Whenever a node learns about a new `BlockHead` it should attempt to import the block - let that be through the `HelloMessage` or through the [block pubsub protocol](data-propagation.md#block-propagation). For that it may fetches the ancestors of that block through bitswap until it is fully caught up. Importing in this context refers to the node confirming that the block is valid, as described in [Validation](./validation.md) and storing it locally. This mode is known as "syncing". During "syncing" the same node may not mine blocks.
+
+
+
+## Establishing a network
+
+Every node should aim to establish a set of stable connections to the network in order to stay up to date with latest changes and block propagation - aka to sync - as well as to participate in the processes of the network - e.g. mining.
+
+In order to do that a node must be "bootstrapped" when starting up. At its simplest, bootstrapping can be thought of as secure network joining. It must enable a new node to get the latest head/chain that the majority of the network power is mining off and must be both:
+
+- Robust — able to do so in adversarial networks (so long as honest nodes make up the majority of the network).
+- Consistent — able to reliably get the canonical chain (even given poor latency or state loss in our initial peer set).
+
+The specific bootstrapping method and heuristic - or a mix of them - is left up to the node implementation, but we provide some example ideas of how this could be done and what aspects should be taken into consideration for that.
+
+### Trusted bootnodes
+
+As with many other networks, trusted entities - the Filecoin foundation, large exchanges, good actors in the community, etc)- might publish a set of bootnodes one can connect to in order to establish network connections.
+
+Keep in mind that, because these could easily be DoS'ed, a) large set of different entities should be pre-shipped but only a few should be tried at a time b) they might drop connections rather quickly after establishing (and ban that peer for a short period of time) in order to serve more nodes. Thus you want to make sure to send a discovery request over the bootnodes to have nodes to connect to.
+
+A node should not attempt to connect to more than 10 bootnodes at a time. And should diversify its set of nodes as quickly as possible.
```

Comment:
number seems a bit arbitrary. May change to "should not attempt to connect to more than a handful of nodes (e.g. 10) at a time: these will be used to quickly diversify a set of nodes to build a network off of"

Reply:
Yes, the number it sort-of arbitrary, but not newly defined. It is the combination of the [first 5 bootnodes](https://github.com/filecoin-project/specs/pull/295/files#diff-caf96f80dec81066bbd3e06dfe6b4d25L54)  and the expansion of [another 5 slots via kademlia](https://github.com/filecoin-project/specs/pull/295/files#diff-caf96f80dec81066bbd3e06dfe6b4d25L78) as previously stated.

I stated that specific number here, to allow for efficient and clearly defined spam-protection rules for anyone running the bootnodes farm: if anyone sees 10 connects over their entire cluster they are totally in their right to refuse any further connection, because that client doesn't handle according to spec. It also makes more clear the expectation to implementers that they shouldn't _and can't_ rely on just bootnodes to create their networks (we've seen that behaviour in the past, where clients wouldn't implement queries but just used bootnodes and then swamped the bootnodes and stalled the network)

Reply:
+1 to picking a number for now.